wasm/jit/jit_amd64_test.go

```diff
@@ -0,0 +1,1441 @@
+//go:build amd64
+// +build amd64
+
+package jit
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"reflect"
+	"sync"
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/require"
+	asm "github.com/twitchyliquid64/golang-asm"
+	"github.com/twitchyliquid64/golang-asm/obj"
+	"github.com/twitchyliquid64/golang-asm/obj/x86"
+
+	"github.com/tetratelabs/wazero/wasm"
+	"github.com/tetratelabs/wazero/wasm/wazeroir"
+)
+
+func newMemoryInst() *wasm.MemoryInstance {
+	return &wasm.MemoryInstance{Buffer: make([]byte, 1024)}
+}
+
+func requireNewBuilder(t *testing.T) *amd64Builder {
+	b, err := asm.NewBuilder("amd64", 128)
+	require.NoError(t, err)
+	return &amd64Builder{eng: nil, builder: b,
+		locationStack:            newValueLocationStack(),
+		onLabelStartCallbacks:    map[string][]func(*obj.Prog){},
+		labelInitialInstructions: map[string]*obj.Prog{},
+	}
+}
+
+func TestAmd64Builder_pushFunctionInputs(t *testing.T) {
+	f := &wasm.FunctionInstance{Signature: &wasm.FunctionType{
+		InputTypes: []wasm.ValueType{wasm.ValueTypeF64, wasm.ValueTypeI32},
+	}}
+	builder := &amd64Builder{locationStack: newValueLocationStack(), f: f}
+	builder.pushFunctionInputs()
+	require.Equal(t, uint64(len(f.Signature.InputTypes)), builder.locationStack.sp)
+	loc := builder.locationStack.pop()
+	require.Equal(t, uint64(1), loc.stackPointer)
+	loc = builder.locationStack.pop()
+	require.Equal(t, uint64(0), loc.stackPointer)
+}
+
+// Test engine.exec method on the resursive function calls.
+func TestRecursiveFunctionCalls(t *testing.T) {
+	eng := newEngine()
+	const tmpReg = x86.REG_AX
+	// Build a function that decrements top of stack,
```

Comment:
this comment seems like overview in nature, like put in the TestRecursiveFunctionCalls godoc?